tests/unit_tests/class_balancing_test.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import torch
+import unittest
+
+from torch.utils.data import Dataset
+
+from super_gradients.training.datasets.balancing_classes_utils import get_repeat_factors
+
+
```

Comment:
I think a comprehensive test that actually runs in DDP is missing.
Please add one (can be done through the ci config - see sanity_tests workflow).

Reply:
Pff.. I agree. I'll try to implement it once we agree on the approach (i.e., is will be redundant if we decide on option 4)